event-based-simulators/main/event_based_simulators.py

```diff
@@ -419,20 +374,91 @@ def __send_event(self, event_fragment, timestamp = None):
             return newTimestamp
 
     def is_in_productionTime(self):
-        #if there are no shiftplans for a device, it should not be affected by production-time
+        # if there are no shiftplans for a device, it should not be affected by production-time
         has_no_shiftplan = True
         for shiftplan in shiftplans:
             if shiftplan.locationId == self.locationId:
                 has_no_shiftplan = False
                 for timeslot in shiftplan.recurringTimeSlots:
                     if timeslot.slotType == "PRODUCTION":
                         now = datetime.utcnow()
-                        start = datetime.strptime(timeslot.slotStart, shiftplan_dateformat)
-                        end = datetime.strptime(timeslot.slotEnd, shiftplan_dateformat)
+                        start = datetime.strptime(timeslot.slotStart, Shiftplan.dateformat)
+                        end = datetime.strptime(timeslot.slotEnd, Shiftplan.dateformat)
                         if start < now and end > now:
                             return True
         return has_no_shiftplan
 
+    # Measurements functions #
+    def measurement_functions(self, measurement_definition, task):
+        MachineSimulator.generate_measurement(self=self, measurement_definition=measurement_definition)
+        MachineSimulator.send_create_measurements(self=self, measurement_definition=measurement_definition)
```

Comment:
the name "send_create_measurements" seems a bit odd; maybe "send_measurement" is better?